### PR TITLE
[14.0][MIG] pos_partner_firstname

### PR DESCRIPTION
**Pull request by iwkse**

### CODE REVIEW

**Comment by iwkse** [comment]
@robyf70 what about merging this PR?

**Review by robyf70 — APPROVED** [review]
LGTM

**Review by francesco-ooops — CHANGES_REQUESTED** [review]
Fields "name" and "surname" should appear only for contacts with company_type = individual and not for companies. 

This is how it looks in v.12:

image

this is how it looks in current PR:

image

**Review by francesco-ooops — CHANGES_REQUESTED** [review]
furthermore if you try to add a product to pos order (just clicking on a product from main POS interface) you get this error: 


Traceback:
TypeError: ProductScreenChildren.find is not a function
    at Object._clickProduct (http://oca-pos-14-0-pr744-93e76fa4b6c5.runboat.odoo-community.org/web/content/479-0bad073/point_of_sale.assets.js:1573:376)
    at ProductItem.__trigger (http://oca-pos-14-0-pr744-93e76fa4b6c5.runboat.odoo-community.org/web/content/474-ac31fe1/web.assets_common.js:1270:9)
    at ProductItem.owl.Component.__trigger (http://oca-pos-14-0-pr744-93e76fa4b6c5.runboat.odoo-community.org/web/content/474-ac31fe1/web.assets_common.js:1422:17)
    at ProductItem.trigger (http://oca-pos-14-0-pr744-93e76fa4b6c5.runboat.odoo-community.org/web/content/474-ac31fe1/web.assets_common.js:1259:33)
    at Object.p488.on.click (eval at _compile (http://oca-pos-14-0-pr744-93e76fa4b6c5.runboat.odoo-community.org/web/content/474-ac31fe1/web.assets_common.js:973:178), <anonymous>:18:117)
    at invokeHandler (http://oca-pos-14-0-pr744-93e76fa4b6c5.runboat.odoo-community.org/web/content/474-ac31fe1/web.assets_common.js:795:144)
    at handleEvent (http://oca-pos-14-0-pr744-93e76fa4b6c5.runboat.odoo-community.org/web/content/474-ac31fe1/web.assets_common.js:799:92)
    at HTMLElement.handler (http://oca-pos-14-0-pr744-93e76fa4b6c5.runboat.odoo-community.org/web/content/474-ac31fe1/web.assets_common.js:801:58)

**Comment by matteoopenf** [comment]
@iwkse will you fix  the last errore?

**Comment by iwkse** [comment]
> @iwkse will you fix the last errore?

Si, going to fix both issues and update the PR

**Comment by matteoopenf** [comment]
> > @iwkse will you fix the last errore?
> 
> 
> 
> Si, going to fix both issues and update the PR

Many thanks

**Comment by iwkse** [comment]
First issue fixed,
the second (ProductScreenChildren) doesn't come from this module but it's an issue with the module pos_reset_search [#705](https://github.com/OCA/pos/issues/705)
@francesco-ooops @matteoopenf

**Comment by matteoopenf** [comment]
> First issue fixed, the second (ProductScreenChildren) doesn't come from this module but it's an issue with the module pos_reset_search [#705](https://github.com/OCA/pos/issues/705) @francesco-ooops @matteoopenf

Hi @iwkse, I try the runboat build and I obtain the same error when I add a product.

This is the traceback:

Traceback:
TypeError: ProductScreenChildren.find is not a function
    at ProductScreen._clickProduct (http://oca-pos-14-0-pr744-b1a9d2d9768b.runboat.odoo-community.org/web/content/479-4b4b4eb/point_of_sale.assets.js:1573:376)
    at ProductItem.__trigger (http://oca-pos-14-0-pr744-b1a9d2d9768b.runboat.odoo-community.org/web/content/474-993f217/web.assets_common.js:1270:9)
    at ProductItem.owl.Component.__trigger (http://oca-pos-14-0-pr744-b1a9d2d9768b.runboat.odoo-community.org/web/content/474-993f217/web.assets_common.js:1422:17)
    at ProductItem.trigger (http://oca-pos-14-0-pr744-b1a9d2d9768b.runboat.odoo-community.org/web/content/474-993f217/web.assets_common.js:1259:33)
    at Object.p488.on.click (eval at _compile (http://oca-pos-14-0-pr744-b1a9d2d9768b.runboat.odoo-community.org/web/content/474-993f217/web.assets_common.js:973:178), <anonymous>:18:117)
    at invokeHandler (http://oca-pos-14-0-pr744-b1a9d2d9768b.runboat.odoo-community.org/web/content/474-993f217/web.assets_common.js:795:144)
    at handleEvent (http://oca-pos-14-0-pr744-b1a9d2d9768b.runboat.odoo-community.org/web/content/474-993f217/web.assets_common.js:799:92)
    at HTMLElement.handler (http://oca-pos-14-0-pr744-b1a9d2d9768b.runboat.odoo-community.org/web/content/474-993f217/web.assets_common.js:801:58)

**Comment by iwkse** [comment]
> pos_reset_search

Hey @matteoopenf , yes that's the bug I was pointing out of pos_reset_search #705 
It's not related to this module but it's an issue with the module pos_reset_search which is loaded in the runboat.

**Comment by matteoopenf** [comment]
> > pos_reset_search
> 
> Hey @matteoopenf , yes that's the bug I was pointing out of pos_reset_search #705 It's not related to this module but it's an issue with the module pos_reset_search which is loaded in the runboat.

go allora LGTM

**Review by matteoopenf — APPROVED** [review]

File: pos_partner_firstname/models/res_partner.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    @api.model
+    def create_from_ui(self, partner):
+        if "is_company" in partner:
+            partner["is_company"] = partner["is_company"] == "true"
+        return super(ResPartner, self).create_from_ui(partner)
+
+    @api.model
+    def get_names_order(self):
+        """Allow POS frontend to retrieve 'partner_names_order'"""
+        return self._get_names_order()
```
Comment (ivantodorovich) [inline pos_partner_firstname/models/res_partner.py:19]:
```suggestion
```
I'd move this method to a `pos.config` technical/hidden field instead.
That way you avoid calling that endpoint on the client form constructor

File: pos_partner_firstname/static/src/js/models.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+odoo.define("pos_partner_firstname.models", function (require) {
+    "use strict";
+
+    var models = require("point_of_sale.models");
```
Comment (ivantodorovich) [inline pos_partner_firstname/static/src/js/models.js:4]:
```suggestion
    const models = require("point_of_sale.models");
```
nitpicking

File: pos_partner_firstname/static/src/js/models.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+    models.load_fields("res.partner", ["is_company", "firstname", "lastname"]);
+});
```
Comment (ivantodorovich) [inline pos_partner_firstname/static/src/js/models.js:7]:
```suggestion

    return models;
});
```

To play nice with other modules

File: pos_partner_firstname/static/src/js/screens.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                    }
+                    $(".client-name").val(name);
+                }
+            }
+
+            updatePerson() {
+                this.display_client_details(this.props.partner);
+            }
+
+            display_client_details(partner) {
```
Comment (ivantodorovich) [inline pos_partner_firstname/static/src/js/screens.js:53]: You shouldn't need to do all this in plain js/jquery here.

You should use `owl` and `qweb` attributes to achieve this, look for example [the `t-on-click` or `t-attf-class` on this line](https://github.com/odoo/odoo/blob/77287edb77fb53836af2aa70c7b14c24e23d6002/addons/point_of_sale/static/src/xml/Screens/ProductScreen/Orderline.xml#L5), should give you a pointer on how to achieve it

Comment (iwkse) [reply pos_partner_firstname/static/src/js/screens.js:53]: @ivantodorovich thank you for the suggestions!

**Comment by francesco-ooops** [comment]
@legalsylvain you can close this one too

**Pull request closed without merging**